internal/querynode/task_scheduler.go

```diff
@@ -17,25 +17,62 @@
 package querynode
 
 import (
+	"container/list"
 	"context"
+	"fmt"
+	"runtime"
 	"sync"
+	"sync/atomic"
+
+	"go.uber.org/zap"
 
 	"github.com/milvus-io/milvus/internal/log"
 )
 
+const (
+	maxReceiveSQChanLen = 1024 * 10
+	maxExecuteSQChanLen = 1024 * 10
+)
+
 type taskScheduler struct {
 	ctx    context.Context
 	cancel context.CancelFunc
 
-	wg    sync.WaitGroup
-	queue taskQueue
+	// for search and query start
+	unsolvedSQTasks *list.List
```

Comment:
limit unsolvedSQTasks size to avoid OOM.
1) we need to removed time out unsolvedSQTasks 
2) if nothing timeout, maybe simply evict the oldest request in queue if reached unsolvedSQTasks limit